chalice/deploy/deployer.py

```diff
@@ -206,6 +312,13 @@ def deploy(self, config, chalice_stage_name=DEFAULT_STAGE_NAME):
             rest api, lambda function, role, etc.
 
         """
+        try:
+            return self._do_deploy(config, chalice_stage_name)
+        except Exception as error:
```

Comment:
Offline: Catch LambdaClientError and botocore.exceptions.ClientError here instead.